src/modelutils.jl

```diff
@@ -981,3 +981,135 @@ function topicdist(model::TopicModel, doc_indices::Vector{<:Integer})
 end
 
 topicdist(model::TopicModel, doc_range::UnitRange{<:Integer}) = topicdist(model, collect(doc_range))
+
+function findcoherence(model::TopicModel, num_topic_words::Integer=20, buffer=Threads.nthreads())
+	"""
+    Calculate the coherence of topic words based on the model's corpus.
+    This algorithm is based on a modified version of the UMass Coherence score.
+    """
+
+    (num_topic_words > 0) || throw(ArgumentError("num_topic_words must be a positive integer."))
+
+    num_topic_words = min(num_topic_words, model.V)
+
+	## Get top topic words
+	topic_words = Matrix{Int}(undef, num_topic_words, model.K)
+	for i in 1:model.K
+        topic_words[:,i] .= model.topics[i][1:num_topic_words]
+    end
+
+	## Create Pair Generator Channel
+	topic_word_pairs =  one2prev_generator(topic_words, model.corp.vocab, buffer)
+
+	## Collect confirmation scores in a results channel
+	coherence_values = Channel{Float64}(buffer) do ch
+        Threads.foreach(topic_word_pairs) do pair
+            confirmation = calculate_confirmation(pair, model.corp)
+            put!(ch, confirmation)
+        end
+    end
+    
+	## Accumulate results channel
+    sum_coherence = 0.0
+    coherence_val_count = 0
+	num_pairs = 0
+    for r in coherence_values
+		num_pairs += 1
+		if isnan(r)
+			continue
+		else
+			sum_coherence += r
+			coherence_val_count += 1
+		end
+    end
+
+	(num_pairs > 0) || throw(error("None of the topic words appear in the vocabulary"))
+	(coherence_val_count > 0) || throw(error("None of the topic words appear in the coprus"))
```

Comment:
I think that if all of the topic words are missing from the vocab or from the corpus, then raising an error is better than returning a zero because it helps diagnose what is happening more clearly.